cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/ACLDatasetKey.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.authorization;
+
+import co.cask.cdap.api.common.Bytes;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Class to generate and manage keys for {@link ACLDataset}. Note: This is a copy of MDSKey from data-fabric, but
+ * we don't want a dependency on data-fabric.
+ */
+final class ACLDatasetKey {
```

Comment:
This is a copy of ``cdap-data-fabric``'s ``MDSKey``, because we do not want an extension to have access to CDAP internal classes